lib/pxcors.js

```diff
@@ -0,0 +1,55 @@
+const { ORIGIN_HEADER, ACCESS_CONTROL_REQUEST_METHOD_HEADER } = require('./utils/constants');
+
+function isPreflightRequest(request) {
+    return request.method.toUpperCase() === 'OPTIONS' && request.get(ORIGIN_HEADER) && request.get(ACCESS_CONTROL_REQUEST_METHOD_HEADER);
+}
+
+function runPreflightCustomHandler(pxConfig, request) {
+    const corsCustomPreflightFunction = pxConfig.CORS_CUSTOM_PREFLIGHT_HANDLER;
+
+    if (corsCustomPreflightFunction) {
+        try {
+            return corsCustomPreflightFunction(request);
+        } catch (e) {
+            pxConfig.logger.debug(`Error while executing custom preflight handler: ${e}`);
+        }
+    }
+
+    return null;
+}
+
+function isCorsRequest(request, pxConfig) {
+    return request.get(ORIGIN_HEADER) && pxConfig.CORS_SUPPORT_ENABLED;
+}
+
+function setCorsBlockHeaders(request, pxConfig, pxCtx) {
+    let corsHeaders = getDefaultCorsHeaders(request);
+    const customCorsHeaders = pxConfig.CORS_CREATE_CUSTOM_BLOCK_RESPONSE_HEADERS;
```

Comment:
Naming - `customCorsHeaders` sounds like an object, not a function. I would change it to something like `createCustomCorsHeaders`.